lib/eslint/flat-eslint.js

```diff
@@ -1212,11 +1212,36 @@ class FlatESLint {
     }
 }
 
+/**
+ * Returns whether flat config should be used.
+ * @param {boolean} [allowFlatConfig] Whether or not to allow flat config.
+ * @returns {Promise<boolean>} Where flat config should be used.
+ */
+async function shouldUseFlatConfig(allowFlatConfig) {
```

Comment:
I'm not sure it makes sense to expose the function with this argument. It was a convenience internally, but I think for external consumption, there would be no need to call `shouldUseFlatConfig()` if `allowFlatConfig` was `false`.

So, I think we should just get rid of the argument here and assume anyone calling `shouldUseFlatConfig()` wants to allow it.

Reply:
yeah, that's fair enough, good catch and thanks!

cleans up the tests a bit too :)

Reply:
fixed now